---
layout: tutorial_page
permalink: /BiCG_2017_module4_genefusions_exploration
title: Bioinformatics for Cancer Genomics 2017 Gene Fusions Tutorial
header1: Bioinformatics for Cancer Genomics 2017
header2: Gene Fusions Tutorial
image: /site_images/CBW_cancerDNA_icon-16.jpg
home: https://bioinformaticsdotca.github.io/bicg_2017
---

# Data Exploration

In the following section we will use the ucsc genome browser's [online blat](http://genome.ucsc.edu/cgi-bin/hgBlat?command=start) to explore a number of example positive and negative fusion transcripts.

## Understanding strand and breakpoint position

### HCC1395 RNA-Seq

#### Example 1: PLA2R1-RBMS1

The following two sequence predicted by defuse represent two distinct splice variants
of the PLA2R1-RBMS1 fusion.

~~~
>771050
AAGATGCAAGAAACTGTGCTGTTTATAAGGCAAACAAAACATTGCTGCCCTTACACTG
TGGTTCCAAACGTGAATGGATATGCAAAATCCCAAGAGATGTGAAACCCAAGATTCCG
TTCTGGTACCAGTACGATGTACCCTGGCTCTTTTATCAGGATGCAGAATACCTTTTTC
ATACCTTTGCCTCAGAATGGTTGAACTTTGAGTTTGTCTGTAGCTGGCTGCACAGTGA
TCTTCTCACAATTCATTCTGCACATGAGCAAGAATTCATCCACAGCAAAATAAAAGCG
|CAACAGGAACAAGATCCTACCAACCTCTACATTTCTAATTTGCCACTCTCCATGGAT
GAGCAAGAACTAGAAAATATGCTCAAACCATTTGGACAAGTTATTTCTACAAGGATAC
TACGTGATTCCAGTGGTACAAGTCGTGGTGTTGGCTTTGCTAGGATGGAATCAACAGA
AAAATGTGAAGCTGTTATTGGTCATTTTAATGGAAAATTTATTAAGACACCACCAGGA
GTTTCTGCCCCCACAGA

>770399
GGATTGGATTTAATAAAAGAAACCCACTGAATGCCGGCTCATGGGAGTGGTCTGATAG
AACTCCTGTTGTCTCTTCGTTTTTAGACAACACTTATTTTGGAGAAGATGCAAGAAAC
TGTGCTGTTTATAAGGCAAACAAAACATTGCTGCCCTTACACTGTGGTTCCAAACGTG
AATGGATATGCAAAATCCCAAGAGATGTGAAACCCAAGATTCCGTTCTGGTACCAGTA
CGATGTACCCTGGCTCTTTTATCAGGATGCAGAATACCTTTTTCATACCTTTGCCTCA
GAATGGTTGAACTTTGAGTTTGTCTGTAGCTGGCTGCACAGTGATCTTCTCACAATTC
ATTCTGCACATGAGCAAGAATTCATCCACAGCAAAATAAAAGC|GCAACAGGAACAAG
ATCCTACCAACCTCTACATTTCTAATTTGCCACTCTCCATGGATGAGCAAGAACTAGA
AAATATGCTCAAACCATTTGGACAAGTTATTTCTACAAGGATACTACGTGATTCCAGT
GGTACAAGTCGTGGTGTTGGCTTTGCTAGATTTTTTTTTTTTTTTGTGAGAAATA
~~~

Web-blatting them will give you the following list of alignments.

~~~
   ACTIONS      QUERY           SCORE START  END QSIZE IDENTITY CHRO STRAND  START    END      SPAN
---------------------------------------------------------------------------------------------------
browser details 770399           389     1   392   576 100.0%     2   -  160832579 160836405   3827
browser details 770399           184   392   576   576 100.0%     2   -  161159310 161159999    690
browser details 770399           155   392   550   576  98.8%    12   -   66628514  66628672    159
browser details 770399            78   392   547   576  75.0%    12   -   94818206  94818361    156
browser details 770399            27   361   391   576  96.6%     2   -    5782745   5783131    387
browser details 770399            22   104   125   576 100.0%     5   +  148444986 148445007     22
browser details 770399            21   438   458   576 100.0%     6   -  158134531 158134551     21
browser details 770399            20   265   284   576 100.0%     5   -   84000186  84000205     20
browser details 770399            20   235   266   576  81.3%     2   +   74143821  74143852     32
browser details 771050           288     1   290   538 100.0%     2   -  160832579 160833890   1312
browser details 771050           241   290   538   538  98.4%    12   -   66628424  66628672    249
browser details 771050           238   290   528   538 100.0%     2   -  161157162 161159999   2838
browser details 771050            83   290   453   538  75.5%    12   +   56965481  56965649    169
browser details 771050            39   462   520   538  83.1%     3   +   29804414  29804472     59
browser details 771050            27   259   289   538  96.6%     2   -    5782745   5783131    387
browser details 771050            27   441   472   538  96.7%     3   +  192743568 192743603     36
browser details 771050            25   461   493   538  76.7%     2   -  182283137 182283166     30
browser details 771050            22     2    23   538 100.0%     5   +  148444986 148445007     22
browser details 771050            21   336   356   538 100.0%     6   -  158134531 158134551     21
browser details 771050            20   163   182   538 100.0%     5   -   84000186  84000205     20
browser details 771050            20     1    24   538  91.7%     1   +  162028403 162028426     24
~~~

Click 'browser' for the first alignment for the alignment to the PLA2R1 gene.

Do you expect the genomic breakpoint to be upstream (on the left)
or downstream (on the right) relative to the aligned sequences?
<div class="hidden">
> upstream (on the left)
</div>
&nbsp;

Go back and click 'browser' for the second alignment for the
alignment to the RBMS1 gene.

Do you expect the genomic breakpoint to be upstream (on the left)
or downstream (on the right) relative to the aligned sequences?
<div class="hidden">
> downstream (on the right)
</div>
&nbsp;

The following sequence is a destruct breakpoint prediction most
likely associated with the fusion.

~~~
>destruct_31240
TGCAAAAGATCTGGAAAAATGCAGTCTGGTATTTACACATAATTTAAGTTCACAGTGC
AACTGCTCCCATAACCCTAGCTGAAACTGTCTCTTCTTAGTCATTTTTAATTTTCCAA
GATAACTTGGCAAAGCTATTGTTGTTGACATAATAAAGACTGGGCAGAAGGCTTACCT
AGCAAAGCCAACACCACGACTTGTACCACTGGAATCACGTAGTATCCTTGTAGAAATA
ACTTGTCCAAATGGTTTGAGCATATTTTCTAGTTCTTGCTCATCCATGGAGAGTGGCA
AATTAGAAATGTAGAGGTTGGTAGGATCTTGTTCCTGTTGCTAAAACAGAAGAGAGTG
TTGTCCATTAATTTCCAACAGAAGGTGAGATATTTATGTTAACACACCTATTTTTATT
AGCTACTTTCTTTGCTCAAGTCCTTTTAAAGTACTCAGAACCTCAGAACACCAAAGTC
ACCCTGGACTCTTGAAAATAGTGTCTGAAGCTTGGACAA[AA]AAAAAGTAATATTAG
AAAATGAATTCATTTTCTGACAAAAAATTATTGGCTCATCCTCTCAGTTATTTACCCT
CTCAGTGATTTATAATTCATTGCATATGTCACATGTATTTGAAAAACAATTCAAGGTA
TCAAGGCATCATTAGTATAAAGATACTGATTTTAGGTATTAGTCTGATTGCTAAGCTT
TAAGCAGTATAAGCTTTCCTTCCCATTCAAATAGAGAGACACAATATAGGACAAAAGA
ATACTACAGAGTGCCCAGTGTTTGACAACTAGAAAATTATCCTTTTGATGAGTTCATG
TCCTTTGCAGGGACATGGATGAAGCTGGAAACCATCAATCTCAGCAAACTAACACATG
AACAGAAAACCAAACACCGCATGTTCTCACTCATAAGTGGGAGCTGAACAATGAGAAC
ACATGGACACAGGGAGGGGAACATCACACACTGGGGCCTGTCGG
~~~

Add the breakpoint sequence to the web blat query, click on the
same alignments as above, and zoom out to see the position of
the breakpoint relative to the fusion.

To view the alignments supporting the fusion in IGV, click
File->Open URL and copy/paste the following URL into
the text box for PLA2R1:

`http://cbwmain.dyndns.info/Module7/bams/genes/HCC1395_PLA2R1.bam`

Then zoom to `chr2:160,832,138-160,833,113`.

You can also add the following URL for RBMS1.

`http://cbwmain.dyndns.info/Module7/bams/genes/HCC1395_RBMS1.bam`

Then zoom to `chr2:161,159,715-161,160,281`.

#### Example 2: RAB7A-LRCH3

Web blat the following three predictions involving LRCH3.

~~~
>2827083
CTCCGAGATCACAAGGTAGAGACACTTTCAACCGGTACTCAATATGTTTACGCAGTTG
GTCTATTAATTCCAGCTCTTCTTCTCTCTGTCTTGAATTCTGTCCTGTTATGGAATCT
GTAGAATCAGTGGTAGGTGCAGCAGATGGAGGAAGGGGTGAAGCATGACCTGTTTCTG
TTGCAGGTGAACTTAATAGAGCATTAGGTCTGTCATCACTC|GCGGCCGCTGCGCTGG
GGGCTCCGGGCCGGGCGCGTCGCGAGGGCTCCCGCCGAGGAGGAGACTAAACGGAGGA
CAGAAGCGAGAAGGTCCAAGTTCTGGTTCCAGGGAACTCTCCCGAGCTCTCCAAGCCG
CCAACTCCGCCGCTGCCGCCGCCTCAGGCTTTATGGCCAAGACTCCAGGCCCGCTCCC
ACTTCCGCCACCGCCG
>2831138
GGAGCAGCTCTAACTGACGGTGTTGTTCTTTGCCATTTGGCCAATCATGTGCGACCTC
GATCTGTCCCAAGCATTCATGTTCCCTCACCAGCTGTACCTAAATTAACAATGGCGAA
ATGCAGGCGAAATGTGGAAAATTTCCTAGAAGCTTGCAGAAAAATTGGTGTACCTCAG
|AGCATAAAAAGATGGAAAGCTTCCAATTCACGCTGCCAAGCATAACTCTGACAGCAA
TTCAACAAAGCAGCACCAAAAAATACACCTACAGGTTAATCTCACTTGTGGAATATAT
AAACACAAAGTAAAAGTCTAGGCTGGCTGTGGCGGCAGGTACCTAGAATCCCAGCTCC
TTGGAGGCTGAGGTGGGAGGATTGCTTGAGCCCAGGAGTTTAAGACCAGCCTGGGCAA
CACAGTGAGAACCCCTGTCTCA
>2831232
GAGCGGGCCTGGAGTCTTGGCCATAAAGCCTGAGGCGGCGGCAGCGGCGGAGTTGGCG
GCTTGGAGAGCTCGGGAGAGTTCCCTGGAACCAGAACTTGGACCTTCTCGCTTCTGTC
CTCCGTTTAGTCTCCTCCTCGGCGGGAGCCCTCGCGACGCGCCCGGCCCGGAGCCCCC
AGCGCAGCGGCCGCG|AGTGATGACAGACCTAATGCTCTATTAAGTTCACCTGCAACA
GAAACAGTTCATCATTCCCCTGCATATTCTTTTCCTGCTGCTATCCAGAGAAATCAGC
CTCAGCGCCCTGAAAGCTTCCTTTTCCGAGCAGGTGTCAGGGCAGAAACCAACAAAGG
TCATGCTTCACCCCTTCCTCCATCTGCTGCACCTACCACTGATTCTACAGATTCCATA
ACAGGACAGAATTCAAGACAGAGAGAAGAAGAGCTGGAATTAATAGACCAACTGCGTA
AACATATTGAGTAC
~~~

For each prediction, is the breakpoint upstream or downstream?
<div class="hidden">
<p>
> 2831232: upstream (to the left)
</p><p>
> 2827083: upstream (to the left)
</p><p>
> 2831138: downstream (to the right)
</p>
</div>
&nbsp;

Is it possible that all three fusion transcripts arrised from the same
breakpoint?
<div class="hidden">
> No
</div>
&nbsp;

Web blat the RAB7A-LRCH3 breakpoint.

~~~
>58737
TCGCATTTTCTGGTATTTTGTATAAATGGGATCTGTATATATACTTTAAGTATACTGT
GTATTTATTTGGGGGGGAGTATCTGGCCTCATTCGACATTATTATTTTATTTGTGCTG
TATGTATCTATAGTTCATTCCTTTTTGTAGCTCAGTGGTATTTCATTGTATAGCTATA
TCACAGTTTGATTAGCCTTTCTGTTGATGGACATTTGGGTTGATTCACGTTTCTGGCT
GTTACAACAAAAGCTGCTGTGAACACTGGTGCACAAGTCTCTGTCTAGATCTGTGTTT
TCATTTCCTTTGGGTAGGTAGCTGTTTCGGAGGGGAATGGTTGGGGCATATGGTAGGC
ATATGCTTTAACTTTGTTTAAAAAGTTGAAATACTTAATACATCATAAAATTTGCCTA
TGTAAAGTGTGCGTGTTTTGTTTTGTTTTGTTTTGTTTTTGGAGACAGGGTCTCATTC
TGTTGCGCAGGCTGGAGTGTAATAGTTTGATCACAGCT[]CAGGAGAATCACTTGAAC
CCGAAAGGCAGAGGTTGTGGTGAGCTGAGAGTGCTCTATTGCACTCTAGCCTGGGCAA
CAAGAGCGAGACTCCATCTCAAAAATAAAAAATAAAAACAGCTGGGCGTGGTGGCTCA
TGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGCAGGCGAATCACCTGAGGTTGGGAG
TTCGAGAACGGCCTGACTAACATGGAGAAACCCCGTATCTACTAAAAATACAAAAAAA
TTAGCTGGGTATGGTGGTGCATGCCTGTAATCCCAGCTACTTGGGAGGCTGAGGCAGG
AGAATTGCTTGAACCCGGGAGGCGGAGGTTGTGGTGAGCTGAGATCGTGCCATTGCAC
TCCAGCCTGGGCAATAAGAGCTAAACTCCATCTCAAAAAAAAAAAAACAAAAAACAAA
AAAACAAAAACTTGGCTTTTAAAAATACAGAATTAATTCATTTCACCAAATCCTCATT
TCTTTTTCTTCTTCTTCTTTTTTTTTTTTTTTT
~~~

Which transcripts does this breakpoint explain?
<div class="hidden">
> 2831232, 2827083
</div>
&nbsp;

## Identifying false positives

### 313B RNA-Seq

#### Example 1

~~~
>19455
CAGGAAATCTTCAGCAAGCTGTCTTACTTCTTTTGGCCAAGTCGCACTCCACATTAGA
GTTTGCCTATCAGGTCTTATTTGATCCACAATCTTCCTTATTTGGGGT|TCAAATCCC
ATATCCAGCATCCTATCAGCTTCATCCAACACTAAGTACTTGCAGAAGTCTAATCC
~~~

Notice that in the alignment summary, the best matches overlap in the query sequence.  In deFuse this prediction is given a lower probability according to the breakpoint homology feature.

~~~
   ACTIONS      QUERY           SCORE START  END QSIZE IDENTITY CHRO STRAND  START    END      SPAN
---------------------------------------------------------------------------------------------------
browser details 19455            132     1   141   171  97.2%    17   +   62499145  62499374    230
browser details 19455             96     2   154   171  88.8%    22   +   38890737  38890976    240
browser details 19455             67   104   171   171 100.0%     Y   -   15027123  15027592    470
browser details 19455             62   108   171   171  98.5%     X   -   73350813  73350876     64
browser details 19455             40    88   141   171  87.1%     6   -   74118977  74119030     54
browser details 19455             32    84   116   171 100.0%    22   -   49607171  49613016   5846
browser details 19455             22    92   114   171 100.0%     4   +   72709432  72709456     25
browser details 19455             21    95   116   171 100.0%     6   +  131014581 131014603     23
browser details 19455             20    93   112   171 100.0%     2   -  157047083 157047102     20
browser details 19455             20   101   122   171  95.5%    14   -   29461851  29461872     22
browser details 19455             20     1    20   171 100.0%     3   +  175393751 175393770     20
~~~

#### Example 2

~~~
>29877
CTGTTTCCTCTTTTACCAAGGACCCGCCAACATGGGCCG|GCTATCTTGTTGCGGAGCTT
CTTGCTGGGGATAATGGCGATCTCCTTGCACACGCGCTTGTTCGTGTG
~~~

Blat the sequence.  Notice in the alignment summary, one part of the query sequence maps to many locations in the genome.  Although one of the alignments may represent a true fusion, the prediction is more likely a mapping artifact.

~~~
   ACTIONS      QUERY           SCORE START  END QSIZE IDENTITY CHRO STRAND  START    END      SPAN
---------------------------------------------------------------------------------------------------
browser details YourSeq           68    40   107   107 100.0%    22   -   32435560  32435627     68
browser details YourSeq           64    40   107   107  97.1%    15   +   82824392  82824459     68
browser details YourSeq           64    40   107   107  97.1%    15   +   83208735  83208802     68
browser details YourSeq           62    40   107   107  95.6%     5   -  116052023 116052090     68
browser details YourSeq           60    40   107   107  94.2%    17   -   29158008  29158075     68
browser details YourSeq           58    40   107   107  92.7%     1   +  167131923 167131990     68
browser details YourSeq           57    44   102   107  98.4%     6   -   63257401  63257459     59
browser details YourSeq           54    44   107   107  92.2%     6   +   50825228  50825291     64
browser details YourSeq           54    41   106   107  85.8%     3   +  143574838 143574900     63
browser details YourSeq           37    44    88   107  91.2%     7   -   37156486  37156530     45
browser details YourSeq           35     1    35   107 100.0%    15   -   82824833  82824867     35
browser details YourSeq           35     1    35   107 100.0%    15   -   83209176  83209210     35
browser details YourSeq           34    40    75   107  97.3%    11   +  110976470 110976505     36
browser details YourSeq           30     8    39   107  96.9%    22   +   32435452  32435483     32
browser details YourSeq           20     1    20   107 100.0%     5   +  144765194 144765213     20
~~~

#### Example 3

~~~
>40571
TAGAATTAGAATTGTGAAGATGATAAGTGTAGAGGGAAGGTTAATAGTTGATATTGCTAG
TGTGGCGCTTCCAATTAGGTGCATGAGTAGGTGGCCTGCAGTAAT|GTTAGCGACAGGGA
GGGATGCGCGCCTGGGTGTAGTTGTGGGGGAGGAAGTGGCTAGCTCAGGGCTTCAGGGGA
CAGACAGGGAGAGATGACTGAG
~~~

Blat the sequence and select the first alignment result.  Ensure you have the NUMT track turned on in UCSC.  This fusion prediction is actually a NUMT insertion in the patient's genome.

#### Example 4

~~~
>33864
CCAGGGCGCCATTGAGCGGCGAGGGGGTGAGGGGGTTGACGGTGGCGGTGGTCCTGGTCG
CGGTGGAAAGCATCCCTAGCGAAGGGGACTTGGGCTCATGGCTCATGCCTG|CACCAGTA
AGGTCTGGTCCGTCCTCCTCCCGGCTGCTCTGCAGACACTGTGCTGGCCTCAGCTCCTGG
GCCATCCTGGGGCCTCTGGGCAG
>17735
CATGGGCACGCGCTTGGGTGTGCTGGCGGGGGAGCTGTGGTTGGTGGCCGGAGAGGACAC
GGGGGACGACTCGCTGCTCAGTGAGGACC|CTGCACCAGTAAGGTCTGGTCCGTCCTCCT
CCCGGCTGCTCTGCAGACACTGTGCTGGCCTCAGCTCCTGGGCCATCCTGGGGCCTCTGG
GCAGGGTCTCCGTGGGGGCGCGTGGCCGGGTCTCGGACT
~~~

Blat both sequences and select the first alignment results.  These are examples of likely read through chimeras.

#### Example 5

~~~
>5655
TGATCAAGCAACTTCCCTGAGGATCCTCAACAATGGTCATGCTTTCAACGTGGAGTTTGA
TGACTCTCAGGACAAAGC|AGAACGTAAGCTCCATGAGGACCAGGAAGTCTGTCTGCTTT
GTTCACTGCTGGATCCCGTGACTCGGAACAGTGCACGTAACAGGTGTTCAATAAACCTTT
GTTGAATGAATAAGTGAA
>11908
TCTGTTTCCTATGATCAAGCAACTTCCCTGAGGATCCTCAACAATGGTCATGCTTTCAAC
GTGGAGTTTGATGACTCTCAGGACAAAGC|AGGGGCTCTTTCCAGGATTCCTGGGTGATG
GTGCATGATTCTAACAAGCAACAACAGAGGATGAACCCCCGGCCAGATTCAGAAAACCCC
ACGCCCCTTCCAGGCA
~~~

Blat both sequences.  Select browser for any alignment result and then browse to `chr1:110,721,056-110,731,655` and `chr8:86,373,054-86,382,253`.  This is an example of rearrangement inducing spurious intronic transcription.

#### Example 6

~~~
>37434
TTGGCATCAAATAGATGAACAGGAGAAAAGCTGTTTTAATGTATGTACTCACAGATGGGA
ATCCCACAAGAATATGAGACTTAAAGAACAGGCCAGGT|TATTCCAGGATCTTTGGAGAC
CCGAGGAAAGCCGTGTTGACCAAAAGCAAGACAAATGACTCACAGAGAAAAAAGATGGCA
GAACCAAGG
>26643
CCGGGACAGTCTGAATCATGTCCTTCAGTAAGCCAGCCCATCTACCAGCTGTTCAGAACC
TGACGGCTTTAGTTGCCCTTGGTTCTGCCATCTTTTTTCTCTGTGAGTCATTTGTCTTGC
TTTTGGTCAACACGGCTTTCCTCGGGTCTCCAAAGATCCTGGAATAACCT|TCCTGGTGG
AGTAGAAGTAGTCTATAGCTTCTCCTTGGTAGTCCAGATGGGTCTCCCCAGCCAATGCAT
AACTCTCTCTTTGCCTTTTGATTCAGAGGCATGTGGAGCTCAGCGTGGCCAGGT
>29118
AGTAAGCCAGCCCATCTACCAGCTGTTCAGAACCT|AGAGGTCTTAGTTCCGGAGGGAGG
AATGCTGCCACCAGGAGACACAACAATGATTCAATTAAACTAGAATTTACGACTGC
>29539
AGGCACACTCAAACAACGACTGGTCCTCACTCACAACTGATAAGGCTTCCTTGATATGAG
CTGCTGGGTCCGGGACAGTCTGAATCATGTCCTTCAGTAAGCCAGCCCATCTACCAGCTG
TTCAGAACCTGACGGCTTTAGTTGCCCTTGGTTCTGCCATCTTTTTTCTCTGTGAGTCAT
TTGTCTTGCTTTTGGTCAACACGGCTTTCCTCGGGTCTC|CAAAGCCATCTTGCTGTTAT
CAACAGCATCGAGTAATGATAGGTATCTGGAATGTTCAATATGACCTGCCGCGCTCCAGG
CGGCGCTCCCCGCCCCTCGCCCTCCGCCTCCGCCTCCGCCTCCTGCTTAGCTCGCGCCTA
CTCG
>29538
TTCAGAACCTGACGGCTTTAGTTGCCCTTGGTTCTGCCATCTTTTTTCTCTGTGAGTCAT
TTGTCTTGCTTTTGGTCAACACGGCTTTCCTCGGGTCTCCAAAGATCCTGGAATA|ACCT
GTCCAGTAGTTCTGTAGCGGAGCAGGGCAGGTCCTACTTCTTCAAAAGCACTCAGTAAAG
GTGGGGAAGTCCTGAGCAACCT
>29535
AGGCTTCCTTGATATGAGCTGCTGGGTCCGGGACAGTCTGAATCATGTCCTTCAGTAAGC
CAGCCCATCTACCAGCTGTTCAGAACCTGACGGCTTTAGTTGCCCTTGGTTCTGCCATCT
TTTTTCTCTGTGAGTCATTTGTCTTGCTTTTGGTCAACACGGCTTTCCTCGGGTCTCCAA
AGATCCTGGAATA|ACCTGCCGCGCCGCGCTCCTCACACCCGCTTTCACCTCCGGGCGGG
GCAGGGGGCATCGGCGGGTCCCAGGCGCCCAGGTTCCCCTCCCCAGCCCGGACCCCGAGC
CGGGACCCTGGTACCGGCGCCGCTCACCTGCCGCGCTCCAGGCGGCGCTCCCCGCCCCTC
GCCCTCCGCCTCCGCCTCCGCCTCCTGCTTAGCTCGCGCCTACTCGGC
>37354
ACCCTCCAAAGCAACATGAAATGAAACCAAACCACAATAACAACCAAATGAAATAAGACT
GACAAGAAGTATGCGGTCATGGCCAATACATGGCT|CGATTTTTTTTTCTTTAACATGCA
CCTTCCTGAGCAAATAAAGGGCTTTTTTCCACCCCTTCCCGCTTGGCTTTAAATGACCAA
AGAATATT
>20051
TGGAATGTTCAATATGACCTGCCGCGCTCCAGGCGGCGCTCCCCGCCCCTCGCCCTCCGC
CTCCGCCTCCGCCTCCTGCTTAGCTCGCGCCTA|CTCCAGCGACTATGGACAGACTTCCA
AGATGAGCCCACGCGTCCCTCAGCAGGATTGGCTGTCTCAACCCCCAGCCAGGGTCACCA
TCAAAATGGAATGTAACCCTAGCCAGGTGAATGGCTCAAGGAACTCTCCTGATGAATGCA
GTGTG
~~~

Blat all the sequences and select browser for any alignment result.  Then navigate to `TMPRSS2` and `ERG`.
This is an example TMPRSS2-ERG fusion.  Notice some predictions are spliced and there are many variants.

### 313B Complex breakpoints (optional)

#### Example 7

~~~
>fusion_27034
ACAGGGACGTCAGGCCACACAGGAGAAGCAGCGCGCCAATTACCACTAGCAACCATATATACCAGAGATGTACCCAGTCTGTGGTCAGGC
AA|CAACTCCACGGGAGGCAAGGCCTGACAGGCTGAAGTCACCTTTGCTTCACATTTTCTGTGCGTCGCCACCTTGCAGACTCTGCACGA
AACGCCTGTCCCATCGATGGTCACCTTA
>breakpoint_86623
TGGCTTTGGGTAAAAACATGCTGTCAGCAGTGTTATTTTGAGTGCTTAAGATGACAAAAACACGTCTTTATTTTGTGATAAGTCATCTAT
GAGATCTATTACTTAAGTAAAAATAAATAAAAAGAAGTCATAGATGTCTAATGACATCTTGACTCTCTTTTGGGNGTTGGGTGCCACCTA
CAGTGTGCAAATTTTCACTAAGAAAGGGAAGTTTTTCTTGCATTCACATCTCTATTGGAAGAAGGGAAAACAATAACACCACGAAAGCAC
CATAATCATTCCTATGCCAGCTTTCTACATTTGAGTTACCTTTCTTCAAAGTGGAAAGGTGGCTCATGCCTGTAATCCCAGCACTTTGGG
AGGCTGAGGCGGGTGGATCACCTGAGGTCAGGAGTTTGAGACCAGCTCGACCAACATGGTGAAACCCCATCTCTACTAAAAATACAAAG
>breakpoint_70139
TTGTGCCCTTGGGCCTGCTGGGGGTAGGACTCCTCTCCCCTTATCCAGTACAGCCTTCAAAAGGACACTGACATTCCCTTCCCCTGTCCC
CAAGGCCCACATTGGTCCTTGCCCCTGCTGTAACTAACCACTCCCTTTTCCTCCCCCATCTCCCTCTAGCGGCGAAACACGGCCCCAGTC
AGGCGCATAGAGCACCTGGTAAGGTGATGCTGGAGCAGGAGGGGGAAGCAGGTAGCTTTGGGAGTAAGGATCTAGATTTCCTGAAGACAA
AAAGGGCATGGCTCTGGAGTGGGCAGTCTAGAGTAGGGGGTACCCAAAGAGATGTCTAGACACTGTCTGTTAACCACCAGGGATCCACCA
AATCTCTGAACCACTCAAAGCAGCGCAGCACTCTNCATGAGCCCAGGGTTTTCCGGTTTCTGGCTCAGCTCCCAATCTCAAACACAAGGC
CTAGAGAAGCACTTAAGTCACCCATCTGACTTAGATAAATGCACCTTTGCAGAGACCCTCACTAAGCTCTGCCTCCCTGGTCTGTCTTGT
CCATCTGAGAAATGGGGATGCCCTGTCCCACTGGTTTTCAGATGATGGTCAGGGAGCCCCATGGGCTCCCCTGACATGCAAGGAGAAGAG
GATGTGTTTAGGGCTCAGTACTTCCCCAGGCCCCCAATAAACACACAATCTAATAACTGCTTGCTGTGTAAGCACAGCTTAAAGACGAGT
TCCTTAGCCTCCAAACTCTTGCTCCTGAAGGACCTTTTCAGCACTGACTTGGGAGGTATCCAAGTGTCAAAGCCAGGAGGACTTGGCTGC
AGCC
~~~

Blat the above sequences and visit the following genomic locations to view the path:

~~~
chr12:10,333,497-10,341,218

chr20:30,520,386-30,535,106

chr12:53,444,452-53,449,049
~~~

### Trinity assembly of simulated reads (optional)

#### Example 8

~~~
>c0_g2_i1 len=2314 path=[553:0-550 1104:551-1144 3516:1145-1762 25:1763-2313]
TTCTCTTACATGTTGGACTTTATAGAATGAAATGCCAGAGATTTGATATTTATTATCTGA
AATAGGTTTAACATTTGGAGAAATGTGTTCAAACACACAACCTTGGAAAGAAAAGCAAGC
TGATAGGTCAGAGAACAGAGGGTGGGCAGAAAACACGATGTGCATAATTTTGCTGCGTGT
TGGGCAAAAAGTTAGAAAACTGGGTTCTTTGGAGAATAAAGAGCTCTCAGATGAGGTAAA
GGTAAAACACAAATAAATTTCCAACCCTTGATGAGGGCCACGAGATATCAATTTTAAATA
TATATGACCACAATATCAATAATGAAGAAATCTTACGTTTTATTGAATACTTTGGCCAGA
TCAGCCCCTTTGAAGGGATCACTAATTTAGGAAAGAAGAACAACAAACCATCATTCACTT
TCCTGCAGGTTCATCGTTTTCCTAATTATTTCTATGGATCCTGCTTTATTCTTTCTGTAA
ATTAAGGGGCAGGAGGAGAGTTTCTATAATAGATCAGCAATGTCTGGTTCTTATCCAAAT
ATTCCACCTATTAAGGATGTAAATTCTCATGGATTGGACAATCACTCTGCAACATTCCTG
AGAATTGAATAATATAATAATCTTGAAAGTCAGCAGAGTGGATTATCCACTTTTTTTTTC
TGGACTTGAACTTGTGAACTCAACAGTAGCACTGCAAAAGAGCAAATGTGCTAAGCACTT
TGCCAGAGTAACCTTCCATGTAGACTTTTCATCTTAAATACACACAACTGAAAACAACTA
CAATTTTTGGAAAATTTTGTACAAACCCGATACTTCTTTTGATTACATATAAATACAAAT
TAGCTATTTTTTTCCTAAAAAGTGGTTATAATAGTAAATAAATACAAAATAAATCTGACC
ATTATACTTCATGTGCTGGGGTTGAACCCATATAAAATGTACAACTAAATACATTTTAAA
TCTTTAAGGAATAATTCTCTGATTAAAATATTTGTTTTCCCAACTTCTTTTCGTAGATAT
AAATATATTTTCAAAATAGCTGTTTTTTTTTCCATTCCATTCCATAAATAAAGTCTTCAT
TGGGAAATATTAAAGTGTCAACTGGGGTGGGGTTTTTTTGTTTGTTTGTTTTTTTCTTTT
TTCTAAAATATATATATATATATATATTTTAGAAAAAAGAAAAAAACAAACAAACAAAAA
AACCCCACCCCAGTTGACACTTTAATATTTCCCAATGAAGACTTTATTTATGGAATGGAA
TGGAAAAAAAAACAGCTATTTTGAAAATATATTTATATCTACGAAAAGAAGTTGGGAAAA
CAAATATTTTAATCAGAGAATTATTCCTTAAAGATTTAAAATGTATTTAGTTGTACATTT
TATATGGGTTCAACCCCAGCACATGAAGTATAATGGTCAGATTTATTTTGTATTTATTTA
CTATTATAACCACTTTTTAGGAAAAAAATAGCTAATTTGTATTTATATGTAATCAAAAGA
AGTATCGGGTTTGTACAAAATTTTCCAAAAATTGTAGTTGTTTTCAGTTGTGTGTATTTA
AGATGAAAAGTCTACATGGAAGGTTACTCTGGCAAAGTGCTTAGCACATTTGCTCTTTTG
CAGTGCTACTGTTGAGTTCACAAGTTCAAGTCCAGAAAAAAAAAGTGGATAATCCACTCT
GCTGACTTTCAAGATTATTATATTATTCAATTCTCAGGAATGTTGCAGAGTGATTGTCCA
ATCCATGAGAATTTACATCCTTAATAGGTGGAATATTTGGATAAGAACCAGACATTGCTG
ATCTATTATAGAAACTCTCCTCCTGCCCCTTAATTTACAGAAAGAATAAAGCAGGATCCA
TAGAAATAATTAGGAAAACGATGAACCTGCAGGAAAGTGAATGATGGTTTGTTGTTCTTC
TTTCCTAAATTAGTGATCCCTTCAAAGGGGCTGATCTGGCCAAAGTATTCAATAAAACGT
AAGATTTCTTCATTATTGATATTGTGGTCATATATATTTAAAATTGATATCTCGTGGCCC
TCATCAAGGGTTGGAAATTTATTTGTGTTTTACCTTTACCTCATCTGAGAGCTCTTTATT
CTCCAAAGAACCCAGTTTTCTAACTTTTTGCCCAACACGCAGCAAAATTATGCACATCGT
GTTTTCTGCCCACCCTCTGTTCTCTGACCTATCAGCTTGCTTTTCTTTCCAAGGTTGTGT
GTTTGAACACATTTCTCCAAATGTTAAACCTATTTCAGATAATAAATATCAAATCTCTGG
CATTTCATTCTATAAAGTCCAACATGTAAGAGAA
~~~

Blat the sequence and select the browser for either match.  Notice the alignment of each end of the sequence to the same loci, ending at a simple repeat.

#### Example 9

~~~
>c24_g1_i1 len=1895 path=[6161:0-730 6959:731-733 1263:734-757 1287:758-1894]
AAAATTAGCCAGGCGTGGTGGCACATGCCTTTAGTTCACACTACTTGGGAGGCTGAGATG
AGAATCACTTGAACCCGGGAGGTGCAGGTTAGAGTGAGCTGAGATTTGTGCCACTGACCT
CCAGCCTGGGAGACAGAGCAAGACACTGTCTCAAAAATAATAATAAGGACGTCCACCACA
CTTTACCTACCAAAATGCCCATTCCCCAAAGATGATTGCTGTTGAACGATTAGGGTACCT
TTCCTGATTTTTTTTGTCTAAGCACATTTAAAAACCAAGACTTTTGGTTTTGAGCGATAG
CTCTGAATCTCCTTTAGGTATGAAGTGTAGGAAATTGTATTTCTGCAGCAAACTTAGGCT
TTGCAACATCAAGTGTGGATTCACTGGAAATTATTTGAAGGAACACATAGCGATCATAAT
AGCCAAAAAATGTTGGGAAATGAAAAAGGGTGGGTTTTTTTAATTAAAAAAAAATTTTTT
TTTTTAAGAGACAGGTTCTTCGCTCCATCACCCAGGATGGAGTACAGTGGCGCGATCATA
GCTCACTGAAGGCTCAAACTCCTGGGCTCAAGCAATCCTGCCTCAGCCTCCCAAGAAGCT
AGGACTACAGGTGTGCACCACCATGCCTAGCTAATGTTTTAATATTTTGTAGAGATGGGG
TCTTGCTGTGTTGCCCTGGCTGGCCTCAAACACCTGGCCTCAAGCAATACAACCACCTCA
GCCTCCCACAGTCCTGGGATTACAGGCATGAGCCACCATGCCCGGCCCCTTATTTTACTT
TATCCTCCTGATTATTTGAATTTGTTACACTAAGCATGTATTTTATAATCAAAAATAATT
AAGATTTAAAACAATGTATTTGACTACTAAAAAGCAAATTATATAAATCAGTTGAATAAT
AATGTTGTTAAAGAGGCTGTTTAGCACTAATTTGACCAAATCTATAGTCAAAGTTTGATT
CTACTTGAATTTTTTGGAAATTTGCTGTTGTATTACTTATTCTTTGCCTTGTTCTCTGTG
AAGTCCTTTCCTAAAAAAGAAACTTATACAATTTAGTTCACAAAACTTTGTTGACCTCTT
TCTCTGTAGAAATGAAAACTAAATATATATAATTGAGAAATAATTCTGATACTCTTCATT
CCTATAAAAGTAATTAAGATAGACCTATAGGATTGTTGAACTTTGCTGCCAAATATAGTA
ACCAGTAACCACAGCTAGCTGTAAAATTGAAAATTCAGTTCAGTCAAACTAGCCATATTT
CAAGTGCTCAACAGCCACTGGTGGCTATTGGTTACCTTACTGACCAGCCAATATAGGCTA
TTTCCGTCATTATAGAAAGCTCTATTGGACAACATTGGTCTAGGACATATATTTTGTATT
TTCCCCTCACCCACCATCCAATAATGAGTAATAGCACAAAGTAAAAGTGACTTAATTTGA
AATTACTTCTGTTTGGCTAGTTTGCTTAGCTGATCACCACCAACATCATAGGTCTATAAA
TCACCCAACTAAGATCAGGGGTGTTTCATGGGAAAATCTACACAAACTATTCATTAAAAA
CCATAGAGCCATGGACTGTCATTTCTGATTTCTGTTTGATTCTTTGTGGTAATCATTGAA
TATACAGGGGCCTCTATACACTCTGGGGAGTATCTATTCTCAGGTGAGCAAAATACCGAA
TACTGAATGGAAATCAACTGTAAGTATTAGGACTTCATATGCAACTTGAATTTTTGGTGC
TGTCCGGAGTCTAGAGCCCCACAATCTGCTTTGGTTACAGTTTATCCCTGTAGGATAAAT
GATCCATTTAACCATTCATCAGAGGTGCTGTAATTTTAAATTGTCTCTTGTCTCCTTCAG
TTAATTTTCAGAATTAAAAACATACCATGGGAAAA
~~~

Blat the sequence and select the browser for either match, then browse to `chr20:18,467,148-18,469,269`.  Notice the alignment of each end of the sequence to the same loci, ending at a SINE repeat.

#### Example 10

~~~
>c19_g1_i1 len=2433 path=[2765:0-2432]
GATCACCTGAGGTCAGGAGTTCGACACCAGCCTGGCCAATGTGGTGAAACCCCATGTCTA
CTAAAAATACAAAAAAATTAGCCAGGCATGGTGGTGTACGCCTGTAATCCCAGCTATTCT
GGAGGCTGAGGCAGGAGAATTGTTTGAATCCAGGAGATGGAGGTTGCAGTGAGCTGAGAT
CGTGCCATTGCACTCCAGCATGGGCAACAGGGCAAGACTCCGTCTCAAAAAAAAGAAAAA
GAGCTGGGGCTGTTGTCCTCAATGGGGGACATTTGAGGACCTAGACTGTCCGAAGTCTTT
CTCTTTGCTCTGGAGGTCACATAGAGCCTGGACCATCCAAGATCTGCTGCTACGGTAAGA
CATCATCTCAGCTTTCCTTTCCCTTCTAGGCCACAGTCCTACCTTCTGAGGTATGGTGGT
GGTCTCAACGAGACTTGTAGATTTTCTGAAAATGACTTTCTCGTTGACTCTAGGGGTTGT
TTCTACCTTGTGGGTGGGGCCTGGTTAATGGTGGGGATGTAGAATGGGGCTCAACAGGCA
GGGCATCCAGATGGTACTGGAAACCCCAAACTCAAGAACTAACTAGGTCAGACCCCCCAG
GAGGCTTATTCTCTTTTTGGAACCCTGGCCCTGGTCCCACCTTCTCTGCACTCGCACTAG
AGGTCGCAGGTGGGATCATGGCACTGGAAGGAAGTAGGTGACTTCTAAGGGCTAAGAGAG
GAAAGAGCAAGGGTACAGGTGGGACCCTAGATGCACCTCTGTGTCCCACCCTCCCTCCCT
GAGTGCACAGCCTTGCCTCTGGGGTGCCCAGGAGGCTGAATTGGGGTTCTAGGTGTGCAG
CCTTGGTTTCCCTTTTGGCCTCTCCTTTGGGCCTAGGCATCATCCAAAGAGACAGCCTCG
TCATTCAGGCTGATGTAGAAGCTGAGGGACTCCCGGAGACCCTCACTGAGAAGAGACTCC
TCCCCTGTGGCAGCTTCAGAAAACAGGAGGGACTGTCCAGCTCTTTCCAGTTGAGTGGTG
TCCTCTGCACAGTCACAGGTAGGGGTATATCCTTGCCAGGGAGCGGGCCAGCCCTCTGCA
GGACACAGGGCTCCTTGAGTAGGCAGCAGATGTCATCCGCCAGCCAGAGTAATGGTCCAC
CAGGGCCTGGAGTGAGGGGAAGGTGAGGCGCGGTGAGATGTACAGCCAGCCATTGTCAAG
GCAGTGGATCCTGTAGTGTCTGATCCGGTCCCAGGATGCAGGGCGGCTGAGGCGGACTGA
CAGAGAGTAAGAGCTCTCCTGGTCTGGCTCTCCCGGATGAGGAAGGCCCCTCCAGGGTTC
CCAGGTAACAACATCAGTTCCTCTGGTTTCTCCCTGCTCAGGCCCTCATACAGCCACCAT
GGGAGACTTTGGCCACGTGGACGCTGGGGATGTTATACTCTCTGCCTGAGACTTCAGACA
GCACCGTCCACCAGTCTCCATCTCAGAGACGATGGTCAATGGCTCCCCGAGTCTCAGCGA
CAGCTCGGGCGGGCCACCTGCCGGGAAACTGCCCAGGGCCACGGCTGTGGCCTTGCTTCT
CCTGCTTCCATGGTCACAGGTCCCTGGCCTTGGACAGAGGAACTCAAGCTTGGGCTTGGC
AGAGATTTTCTTCTGCTGGGCAGACTTCCCATTGTTCCTCAGCAGAGCACTCAGAAGCAC
ATCATCGAGGGAAATTGCTGGGAGGGCCGGTAGGGCGCCATGGGCCGCTGCCCCATCATG
CTGCTCCCCTGGCTGCCCTGCCCCATCATGGCGATGGACGACTGGCCCTGGTAGTGCTGG
CTGCCGCCCTGCGCCGAGCTGTAGTGCGACGTGGCCGCCTGCTGCTGCATCATGGAGATG
GGTTGGACTGCATGTTGATGTTGGTCCGAGACACGTAGTTGCCGATGGTGCCTTGCCCCT
GCATGGGGACGTCCTGCGAGGCGGGTCCGGCGTGGCTGTAGCCGGGCCCAGAGATGCTCA
TGGAGGTGGTGGGCAGCGTGTTAGGCGCCGTCTGCTGCATGGACACGTGGCTCGGCCGTT
GCCAATCTGGCCCTGCAGGAGGGAGGAGGGTGGCAGGCCCGTGCGGATGGCGTCACTCAG
GCTGCCCTGAGAGTGCAGGCCCTGGCTGGAGCCGCTCTGAGTCAGGGCTCCAGGGCCCAG
GTTCATGTTCTGCGTGGGCGGGCAGGAAGCAGGGACTGCATGTTCTGGTTGGAGTCTGCG
ATCGTGGCCAGGTATACCAGGTTCCGGTGCAGGATCTGCTGGTACGCGTGCACTCGGCCG
TCTTGCCCTTGCTCTGGTACTCCAGGATGCACTGGATCAGGTGGTGGTTCTCGTCCAGCA
TTTCTGGATGGTTTGCTGCGTAACCTCCCCTTTGCCTCTTGGCCGGGCAGACGCGAAGGC
CACGGACATGGTGGCGGCGCGGGGCTCAGCCCG
~~~

Blat the sequence and select the browser for either match, then browse to `chr20:35,241,397-35,269,781` and `chr20:60,718,853-60,738,677`.  Notice the alignment of each end of the sequence to the same loci, ending at a SINE repeat.


